pysatMadrigal/tests/test_methods_general.py

```diff
@@ -323,21 +342,29 @@ def test_download_valueerror(self, in_key, in_val, test_verr):
         del self.kwargs['supported_tags'], self.kwargs['kindats']
         self.kwargs[in_key] = in_val
 
-        with pytest.raises(ValueError) as verr:
-            general.download([], **self.kwargs)
-
-        assert str(verr).find(test_verr) >= 0
+        # Get the expected error message and evaluate it
+        eval_bad_input(general.download, ValueError, test_verr,
+                       input_args=[[]], input_kwargs=self.kwargs)
         return
 
     def test_get_remote_filenames_bad_date_array(self):
         """Test raises ValueError for unexpected date_array input."""
         del self.kwargs['supported_tags'], self.kwargs['kindats']
         self.kwargs['date_array'] = []
 
-        with pytest.raises(ValueError) as verr:
-            general.get_remote_filenames(**self.kwargs)
+        # Get the expected error message and evaluate it
+        eval_bad_input(general.get_remote_filenames, ValueError,
+                       "unknown date_array supplied", input_kwargs=self.kwargs)
+        return
 
-        assert str(verr).find("unknown date_array supplied") >= 0
+    def test_convert_pandas_to_xarray_bad_data_vars(self):
```

Comment:
May want to consider additional testing for the underlying method. Seems like pandas to xarray could have a variety of challenges.

Reply:
Please make this an issue.